vm/src/builtins/set.rs

```diff
@@ -71,9 +72,18 @@ impl PySet {
 }
 
 #[pyclass(module = false, name = "frozenset", unhashable = true)]
-#[derive(Default)]
 pub struct PyFrozenSet {
     inner: PySetInner,
+    hash: PyRwLock<PyHash>,
```

Comment:
This could probably be an atomic instead of a lock - you can look at `PyStr` to see how it does it (using `rustpython_common::atomic::PyAtomic<PyHash>`)

Reply:
Good Point. Updated to use a PyAtomic instead. Also replaced the naked constant -1 with hash::SENTINEL